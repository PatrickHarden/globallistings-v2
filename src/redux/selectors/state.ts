import { State } from '../../types/state';
import { theme } from '../../themes/default/theme';
import { createSelector } from 'reselect';
import { FilterType } from '../../types/config';

// State
export const entityMetaSelector = (state: State) => state.entitiesMeta;
export const entitiesSelector = (state: State) => state.entities;
export const uiSelector = (state: State) => state.ui;
export const locationSelector = (state: State) => state.router.location;

// Entities
export const propertiesSelector = createSelector(
  entitiesSelector,
  entities => entities.properties
);

export const geocodeSelector = createSelector(
  entitiesSelector,
  entities => entities.geocode
);

export const propertySearchesSelector = createSelector(
  entitiesSelector,
  entities => entities.propertySearches
);

export const configSelector = createSelector(
  entitiesSelector,
  entities => entities.config
);

export const autocompleteSelector = createSelector(
  entitiesSelector,
  entities => entities.autocomplete
);

export const autocompleteSearchesSelector = createSelector(
  entitiesSelector,
  entities => entities.autocompleteSearches
);

// Meta
export const propertiesMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.propertySearches
);

export const geocodeMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.geocode
);

export const configMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.config
);

export const autocompleteSearchesMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.autocompleteSearches
);

// Config
export const localeSelector = createSelector(
  configSelector,
  config => config.language
);

export const messagesSelector = createSelector(
  configSelector,
  config => config.i18n
);

export const filtersSelector = createSelector(
  configSelector,
  config => config.filters
);

export const rangeFiltersSelector = createSelector(
  filtersSelector,
  filters => filters.filter(f => f.type === FilterType.range)
);

// Other

// TODO add logic to switch between themes based on config
export const themeSelector = (state: State) => theme;

// TODO figure out where it will be coming from
export const googleMapsApiKeySelector = () => 'AIzaSyCzEw2NaQFixR7sZCLnZ1oGQljW7AxLZ7k';
