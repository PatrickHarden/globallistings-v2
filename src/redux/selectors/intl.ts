import { createSelector } from 'reselect';
import { IntlProvider } from 'react-intl';
import { Dictionary } from '../../types/common/dictionary';
import { localeSelector, messagesSelector } from './state';

export const intlSelector = createSelector(
  localeSelector,
  messagesSelector,
  (locale: string, messages: Dictionary<string>) =>
    new IntlProvider({ locale, messages }).getChildContext().intl
);
