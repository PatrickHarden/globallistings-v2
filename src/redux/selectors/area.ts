import { createSelector } from 'reselect';

import { Bounds, LatLng } from '../../types/state';
import { DistanceUnit } from '../../types/config';

import { googleBoundsToBounds } from '../../utils/bounds';
import { haversineRadius } from '../../utils/haversine';
import { paramsSelector } from './config/config';
import { defaultFiltersParamsSelectors } from './config/filters';

import { plpLocationRadiusSelector, plpLocationSelector, locationParamsSelector } from './location';

interface BaseArea {
  readonly bounds: Bounds;
  readonly center: LatLng;
}

export interface RectangleArea extends BaseArea {
  readonly type: 'rectangle';
}

export interface CircleArea extends BaseArea {
  readonly radius: number;
  readonly type: 'circle';
}

export type Area = RectangleArea | CircleArea;

const defaultRadiusSelector = createSelector(
  defaultFiltersParamsSelectors,
  ({ radius }) => {
    if (Array.isArray(radius)) {
      throw new Error('radius can not be an array');
    }

    return radius;
  }
);

const radiusSelector = createSelector(
  defaultRadiusSelector,
  plpLocationRadiusSelector,
  (defaultRadius, radius) => (radius ? radius : defaultRadius)
);

const radiusTypeSelector = createSelector(
  paramsSelector,
  params => (params && params.RadiusType === 'Kilometers' ? DistanceUnit.km : DistanceUnit.miles)
);

export const areaSelector = createSelector(
  radiusSelector,
  plpLocationSelector,
  radiusTypeSelector,
  (radius, location, radiusType): Area => {
    if (!radius || radius === '' || radius === '0') {
      return {
        type: 'rectangle',
        bounds: location.bounds,
        center: location.center,
      };
    }

    const resultRadius = haversineRadius(
      location.bounds,
      location.center,
      Number.parseFloat(radius),
      radiusType
    );

    const boundsCircle = new google.maps.Circle({
      center: location.center,
      radius: resultRadius * 1000,
    });

    return {
      type: 'circle',
      radius: resultRadius,
      bounds: googleBoundsToBounds(boundsCircle.getBounds()),
      center: location.center,
    };
  }
);
