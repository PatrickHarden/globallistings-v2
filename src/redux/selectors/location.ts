import { createSelector } from 'reselect';
import { locationSelector, geocodeSelector } from './state';
import { parseUrl } from 'query-string';
import { plpUrlPathParamsSelector, pathParamsSelector } from './match';

export const queryStringSelector = createSelector(
  locationSelector,
  location => location.search
);

export const pathSelector = createSelector(
  locationSelector,
  location => location.pathname
);

export const locationQuerySelector = createSelector(
  queryStringSelector,
  queryString => parseUrl(queryString).query
);

export const querySelector = createSelector(
  queryStringSelector,
  location => parseUrl(location).query
);

export const locationParamsSelector = createSelector(
  locationQuerySelector,
  pathParamsSelector,
  (query, params) => ({ ...query, ...params })
);

export const plpLocationNameSelector = createSelector(
  plpUrlPathParamsSelector,
  params => params.location
);

export const plpTransactionTypeSelector = createSelector(
  plpUrlPathParamsSelector,
  params => params.transactionType
);

export const plpLocationSelector = createSelector(
  plpLocationNameSelector,
  geocodeSelector,
  (locationName, geocode) => {
    const location = geocode[locationName];
    if (!location) {
      throw new Error('Failed retrieving geocoding for the location');
    }

    return location;
  }
);

export const plpLocationRadiusSelector = createSelector(
  locationQuerySelector,
  ({ radius }) => {
    if (Array.isArray(radius)) {
      throw new Error('radius can not be an array');
    }

    return radius;
  }
);
