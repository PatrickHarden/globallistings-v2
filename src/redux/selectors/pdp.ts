import { createSelector } from 'reselect';
import { InjectedIntl } from 'react-intl';

import { pdpUrlPathParamsSelector, pdpMatchSelector } from './match';
import { propertiesSelector } from './state';
import { intlSelector } from './intl';
import { Location, Property, Description } from '../../types/property';

import { mapAddress } from '../../utils/propertyData/mapAddress';
import { mapImage } from '../../utils/propertyData/mapImage';
import { mapPrice } from '../../utils/propertyData/mapPrice';
import { mapSize } from '../../utils/propertyData/mapSize';
import { mapSpecification } from '../../utils/propertyData/mapSpecification';

import { isCommercialSelector } from './property-search-results';

export interface PropertyDetails {
  readonly address: string;
  readonly id: string;
  readonly imgUrl: string;
  readonly location: Location;
  readonly description?: Description;
  readonly price?: string;
  readonly size?: string;
  readonly specification?: string[];
}

const toPropertyDetails = (
  p: Property,
  intl: InjectedIntl,
  isCommercial: boolean
): PropertyDetails => ({
  address: mapAddress(p, intl, 'AddressSummaryLong'),
  description: p.description,
  id: p.id,
  imgUrl: mapImage(p),
  location: p.location,
  price: mapPrice(p, intl),
  size: mapSize(p, intl, isCommercial),
  specification: mapSpecification(p, intl),
});

export const propertyIdSelector = createSelector(
  pdpUrlPathParamsSelector,
  params => params.propertyId
);

export const propertySelector = createSelector(
  propertiesSelector,
  propertyIdSelector,
  (properties, id) => properties[id]
);

export const propertyDetailsSelector = createSelector(
  propertySelector,
  intlSelector,
  isCommercialSelector,
  (property, intl, isCommercial) => property && toPropertyDetails(property, intl, isCommercial)
);

export const sectionHeadingsSelector = createSelector(
  intlSelector,
  intl => ({
    details: intl.formatMessage({ id: 'OverviewTitle' }),
    specification: intl.formatMessage({ id: 'PdpSpecificationTitle' }),
  })
);

export const isPdpPath = createSelector(
  pdpMatchSelector,
  match => !!match
);
