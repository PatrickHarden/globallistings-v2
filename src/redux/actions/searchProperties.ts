import axios from 'axios';
import { createStandardAction } from 'typesafe-actions';

import { normaliseApiResponse } from '../../api/normalisers';
import { State } from '../../types/state';
import { NormalisedSearchResult } from '../../types/results';
import {
  createRequestPending,
  createRequestFailure,
  createRequestSuccess,
} from '../../types/common/action-creators';
import {} from '../selectors/property-search-results';
import {
  propertiesSearchRequestQuerySelector,
  propertySearchRequestQuerySelector,
} from '../selectors/search-api-request';
import { plpQueryHashSelector } from '../selectors/plp';
import { stringify } from 'query-string';
import { propertyIdSelector, propertyDetailsSelector } from '../selectors/pdp';
import { docToProperty } from '../../api/normalisers/doc-to-property-mapper';
import { Property } from '../../types/property';
import { Dictionary } from '../../types/common/dictionary';

export const searchPropertiesSuccess = createRequestSuccess<
  NormalisedSearchResult,
  'SEARCH_PROPERTIES_SUCCESS'
>('SEARCH_PROPERTIES_SUCCESS');
export const searchPropertiesRequest = createRequestPending('SEARCH_PROPERTIES_REQUEST');
export const searchPropertiesFailure = createRequestFailure('SEARCH_PROPERTIES_FAILURE');

export const fetchProperties = () => (dispatch: Function, getState: () => State) => {
  const state = getState();
  const params = propertiesSearchRequestQuerySelector(state);
  const hash = plpQueryHashSelector(state);
  dispatch(searchPropertiesRequest(hash));

  return axios
    .get('/api/propertylistings/query', {
      params,
      paramsSerializer: stringify,
    })
    .then(res => {
      // TODO Update to dynamic locale once we have that setup
      const normalisedData = normaliseApiResponse('en-GB')(res.data);
      dispatch(searchPropertiesSuccess(normalisedData, hash));
    })
    .catch(ex => {
      dispatch(searchPropertiesFailure(ex, hash));
    });
};

export const searchPropertyRequest = createStandardAction('SEARCH_PROPERTY_REQUEST');
export const searchPropertySuccess = createStandardAction('SEARCH_PROPERTY_SUCCESS')<
  Dictionary<Property>
>();
export const searchPropertyFailure = createStandardAction('SEARCH_PROPERTY_FAILURE')<Error>();

export const fetchPropertyIfRequired = () => (dispatch: Function, getState: () => State) => {
  const state = getState();
  const propertyId = propertyIdSelector(state);
  const params = propertySearchRequestQuerySelector(state);
  const isPropertyDetailsLoaded = propertyDetailsSelector(state);
  if (isPropertyDetailsLoaded) {
    return;
  }

  dispatch(searchPropertyRequest());

  return axios
    .get(`/api/propertylisting/${propertyId}`, { params })
    .then(res => {
      // TODO Update to dynamic locale once we have that setup
      const normalisedData = docToProperty('en-GB')(res.data.Document);
      if (!normalisedData) {
        const err = Error('Failed mapping property');
        return dispatch(searchPropertyFailure(err));
      }
      dispatch(searchPropertySuccess({ [propertyId]: normalisedData }));
    })
    .catch(ex => {
      dispatch(searchPropertyFailure(ex));
    });
};
