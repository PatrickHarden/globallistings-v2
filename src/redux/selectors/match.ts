import { createMatchSelector, RouterRootState } from 'connected-react-router';
import { PlpPath } from '../../types/plp';
import { PdpPath } from '../../types/pdp';
import RoutePaths from '../../constants/routePaths';
import { createSelector } from 'reselect';

export const plpMatchSelector = createMatchSelector<RouterRootState, PlpPath>(RoutePaths.plp);
export const pdpMatchSelector = createMatchSelector<RouterRootState, PdpPath>(RoutePaths.pdp);

export const pathParamsSelector = createSelector(
  plpMatchSelector,
  pdpMatchSelector,
  (plpMatch, pdpMatch) => ({ ...(plpMatch || {}), ...(pdpMatch || {}) })
);

export const plpUrlPathParamsSelector = createSelector(
  plpMatchSelector,
  match => {
    if (!match) {
      throw new Error('Url path must be defined to use this selector');
    }
    return match.params;
  }
);

export const pdpUrlPathParamsSelector = createSelector(
  pdpMatchSelector,
  match => {
    if (!match) {
      throw new Error('Url path must be defined to use this selector');
    }
    return match.params;
  }
);
