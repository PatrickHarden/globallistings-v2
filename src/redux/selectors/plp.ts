import { createSelector } from 'reselect';
import {
  geocodeMetaSelector,
  propertiesMetaSelector,
  propertySearchesSelector,
  configSelector,
} from './state';
import { propertiesSearchRequestQuerySelector } from './search-api-request';
import { murmur2 } from 'murmurhash-js';
import { plpLocationNameSelector } from './location';
import { otherConfigsSelector } from './config/config';
import { plpUrlPathParamsSelector, plpMatchSelector } from './match';

export const plpQueryHashSelector = createSelector(
  propertiesSearchRequestQuerySelector,
  query => murmur2(JSON.stringify(query)).toString()
);

export const plpPropertyRequestStatus = createSelector(
  plpQueryHashSelector,
  propertiesMetaSelector,
  (hash, propertiesMeta) => {
    const propertyMeta = propertiesMeta[hash];
    if (propertyMeta === undefined) {
      return 'NONE';
    }

    return propertyMeta.status;
  }
);

export const propertySearchSelector = createSelector(
  propertySearchesSelector,
  plpQueryHashSelector,
  (propertySearches, hash) => {
    const result = propertySearches[hash];

    if (result) {
      return result;
    }

    return [];
  }
);

const plpLocationGeocodeStatus = createSelector(
  plpLocationNameSelector,
  geocodeMetaSelector,
  (location, meta) => {
    const entityMeta = meta[location];
    return entityMeta ? entityMeta.status : 'NONE';
  }
);

export const isPlpLocationGeocoded = createSelector(
  plpLocationGeocodeStatus,
  status => status === 'SUCCESS'
);

export const isUsageTypeConfigLoaded = createSelector(
  configSelector,
  plpUrlPathParamsSelector,
  (config, path) => {
    if (!config.currentConfig) {
      return true;
    }

    return path.usageType === config.currentConfig;
  }
);

export const usageTypeSelector = createSelector(
  plpUrlPathParamsSelector,
  params => params.usageType
);

export const configEndpointForCurrentUsageTypeSelector = createSelector(
  usageTypeSelector,
  otherConfigsSelector,
  (usageType, configs) => {
    const config = configs.find(c => c.name === usageType);
    if (!config) {
      throw new Error('Config is missing data for config endpoint mapping');
    }

    return config.endpoint;
  }
);

export const isPlpPath = createSelector(
  plpMatchSelector,
  match => !!match
);
