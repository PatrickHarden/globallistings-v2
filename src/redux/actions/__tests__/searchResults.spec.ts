jest.mock('../../selectors/match');

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import residential from '../../../tests/fixtures/residential-response.json';
import { fetchProperties } from '../searchProperties';
import { DeepPartial } from '../../../types/common/deep-partial';
import { State } from '../../../types/state';
import { plpUrlPathParamsSelector } from '../../selectors/match';

const MOCK_HASH = 'MOCK_HASH';
jest.mock('murmurhash-js', () => ({ murmur2: () => MOCK_HASH }));
((plpUrlPathParamsSelector as unknown) as jest.Mock).mockReturnValue({
  location: 'london',
  usageType: 'office',
  transactionType: 'sale',
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const mock = new MockAdapter(axios);
const initState: DeepPartial<State> = {
  router: { location: { search: '?foo=zap' } },
  entities: {
    config: { params: {}, filters: [] },
    geocode: {
      london: {
        bounds: {
          ne: { lat: 1, lng: 2 },
          sw: { lat: 1, lng: 2 },
        },
        center: {
          lat: 1,
          lng: 2,
        },
      },
    },
  },
};

describe('Making an api request for property listings', () => {
  let store: any;
  afterEach(() => {
    mock.reset();
  });

  beforeEach(() => {
    store = mockStore(initState);
  });

  it.only('should dispatch correct actions for successful request', async () => {
    // Arrange
    const apiWithAnyQuery = /\/api\/propertylistings.*/;
    mock.onGet(apiWithAnyQuery).replyOnce(200, residential);

    // Act
    await store.dispatch(fetchProperties());

    // Assert
    const actions = store.getActions();
    expect(actions).toEqual([
      { meta: { hash: MOCK_HASH }, type: 'SEARCH_PROPERTIES_REQUEST' },
      {
        meta: { hash: MOCK_HASH, receivedAt: expect.any(Number) },
        payload: expect.any(Object),
        type: 'SEARCH_PROPERTIES_SUCCESS',
      },
    ]);
    expect(actions[1].payload).toMatchSnapshot();
  });

  it('should dispatch correct actions for network error', async () => {
    // Arrange
    mock.onAny().networkError();

    // Act
    await store.dispatch(fetchProperties());

    // Assert
    const actions = store.getActions();
    expect(actions).toEqual([
      { meta: { hash: MOCK_HASH }, type: 'SEARCH_PROPERTIES_REQUEST' },
      {
        meta: { hash: MOCK_HASH },
        payload: expect.any(Error),
        type: 'SEARCH_PROPERTIES_FAILURE',
      },
    ]);
  });

  it('should dispatch correct actions for network timeout', async () => {
    // Arrange
    mock.onAny().timeout();

    // Act
    await store.dispatch(fetchProperties());

    // Assert
    const actions = store.getActions();
    expect(actions).toEqual([
      { meta: { hash: MOCK_HASH }, type: 'SEARCH_PROPERTIES_REQUEST' },
      {
        meta: { hash: MOCK_HASH },
        payload: expect.any(Error),
        type: 'SEARCH_PROPERTIES_FAILURE',
      },
    ]);
  });
});
