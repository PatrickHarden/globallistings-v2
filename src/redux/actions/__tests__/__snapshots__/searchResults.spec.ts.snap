// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Making an api request for property listings should dispatch correct actions for successful request 1`] = `
Object {
  "entities": Object {
    "properties": Object {
      "GB-ReapIT-cbrrps-CAL180002": Object {
        "address": Object {
          "Country": "GB",
          "Line1": "Ariel House",
          "Line2": "144 Vaughan Way",
          "Locality": "London Dock",
          "PostCode": "E1W",
        },
        "charges": Array [
          Object {
            "amount": 595,
            "amountKind": undefined,
            "chargeKind": "Rent",
            "chargeModifier": undefined,
            "currencyCode": "GBP",
            "interval": "Weekly",
            "onApplication": undefined,
            "perUnit": "Whole",
            "taxModifier": "None",
            "year": undefined,
          },
        ],
        "contacts": Array [
          Object {
            "email": "+44 (0) 20 7205 2201",
            "name": "Canary Wharf Lettings",
            "phone": "canarywharf.lettings@cbre.com",
          },
        ],
        "description": Object {
          "longDescription": "The long awaited London Dock development has finally arrived and you can at last secure your place in it. 

Situated in the heart of the capital, just moments from the extremely desirable St Katharine Docks, this exciting new destination offers apartments and penthouses with the finest level of residents' facilities, a private suite of hotel style amenities including state-of- the-art gymnasium, swimming pool, squash court, virtual golf, treatment room, sauna, steam room, private screening room and residents lounge complimented by 24 hour concierge. 

The brand new third floor one bedroom flat is finished to an immaculate standard. The open plan kitchen and living room is a great size and has attractive wooden flooring as well as all the modern conveniences. The floor to ceiling window allows plenty of light into the flat and access to a huge private balcony which overlooks the square. There is also a wall mounted flat screen television.

The bedroom has a built in double wardrobe offering plenty of storage and meaning the ample space available is not compromised.

The bathroom is a joy and has underfloor heating and there is storage in the hallway which houses the washer dryers and flat controls meaning that they do not hinder the aesthetic or practical enjoyment of the property.

Surrounded by  beautifully landscaped public spaces, shops, bars and restaurants. Tower Hill underground is a short walk away meaning that there is easy access to the District and Circle Line. This makes trips to Chelsea, Victoria and Westminster a short affair. a Tower Hill Gateway DLR is also within walking distance so London Dock offers quick and easy access to the main financial centres of the City and Canary Wharf, as well as the exciting cultural and creative neighbourhoods of Shoreditch, Whitechapel and Bermondsey. The location will work fantastically for students and professionals.

For more information about our fees please visit our web-site's 'Tenant Fees' page located under the Rent tab.",
          "strapline": "An amazing one bed to rent in London&#039;s newest exclusive development, E1W",
        },
        "highlights": Array [
          "Golf Room",
          "Balcony",
          "Gym",
          "Furnished",
          "Swimming Pool",
          "24 hour Concierge",
        ],
        "id": "GB-ReapIT-cbrrps-CAL180002",
        "info": Object {
          "energyPerformance": Array [
            "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/6db2de5d/CAL180002_05.png",
          ],
          "floorPlans": Array [
            "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/39549074/CAL180002_22.jpg",
          ],
        },
        "location": Object {
          "lat": 51.505794,
          "lon": -0.060134,
        },
        "numberOfBedrooms": 1,
        "photos": Object {
          "hero": Object {
            "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/a860e9a6/CAL180002_06_Photo_1_large.jpg",
            "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/a860e9a6/CAL180002_06_Photo_1_small.jpg",
          },
          "main": Array [
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/a860e9a6/CAL180002_06_Photo_1_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/a860e9a6/CAL180002_06_Photo_1_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/cdc6e57e/CAL180002_07_Photo_2_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/cdc6e57e/CAL180002_07_Photo_2_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/c48f7da7/CAL180002_09_Photo_3_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/c48f7da7/CAL180002_09_Photo_3_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/1bc45f97/CAL180002_08_Photo_4_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/1bc45f97/CAL180002_08_Photo_4_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/342bd9b2/CAL180002_10_Photo_5_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/342bd9b2/CAL180002_10_Photo_5_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/be916afa/CAL180002_11_Photo_6_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/be916afa/CAL180002_11_Photo_6_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/ddeac696/CAL180002_12_Photo_7_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/ddeac696/CAL180002_12_Photo_7_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/cbb349dc/CAL180002_13_Photo_8_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/cbb349dc/CAL180002_13_Photo_8_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/500cd96d/CAL180002_14_Photo_9_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/500cd96d/CAL180002_14_Photo_9_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/e9e1ed42/CAL180002_15_Photo_10_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/e9e1ed42/CAL180002_15_Photo_10_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/83d0dae7/CAL180002_16_Photo_11_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/83d0dae7/CAL180002_16_Photo_11_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/32a4fc4b/CAL180002_17_Photo_12_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/32a4fc4b/CAL180002_17_Photo_12_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/4ac16bc3/CAL180002_18_Photo_13_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/4ac16bc3/CAL180002_18_Photo_13_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/cff83809/CAL180002_19_Photo_14_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/cff83809/CAL180002_19_Photo_14_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/e96bcaca/CAL180002_20_Photo_15_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/e96bcaca/CAL180002_20_Photo_15_small.jpg",
            },
            Object {
              "large": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/bfff6f9a/CAL180002_21_Photo_16_large.jpg",
              "small": "/resources/fileassets/GB-ReapIT-cbrrps-CAL180002/bfff6f9a/CAL180002_21_Photo_16_small.jpg",
            },
          ],
        },
      },
    },
  },
  "result": Array [
    "GB-ReapIT-cbrrps-CAL180002",
  ],
}
`;
